src/libfuncs/array.rs

```diff
@@ -1177,13 +1179,199 @@ where
     Ok(())
 }
 
+/// Generate MLIR operations for the `span_from_tuple` libfunc.
+pub fn build_span_from_tuple<'ctx, 'this, TType, TLibfunc>(
+    context: &'ctx Context,
+    registry: &ProgramRegistry<TType, TLibfunc>,
+    entry: &'this Block<'ctx>,
+    location: Location<'ctx>,
+    helper: &LibfuncHelper<'ctx, 'this>,
+    metadata: &mut MetadataStorage,
+    info: &SignatureAndTypeConcreteLibfunc,
+) -> Result<()>
+where
+    TType: GenericType,
+    TLibfunc: GenericLibfunc,
+    <TType as GenericType>::Concrete: TypeBuilder<TType, TLibfunc, Error = CoreTypeBuilderError>,
+    <TLibfunc as GenericLibfunc>::Concrete: LibfuncBuilder<TType, TLibfunc, Error = Error>,
+{
+    // tuple to array span (t,t,t) -> &[t,t,t]
+
+    if metadata.get::<ReallocBindingsMeta>().is_none() {
+        metadata.insert(ReallocBindingsMeta::new(context, helper));
+    }
+
+    let struct_type_info = registry.get_type(&info.ty)?;
+
+    let struct_ty = registry.build_type(context, helper, registry, metadata, &info.ty)?;
```

Comment:
I think we should add a check to ensure that every struct member has the same type.
The Cairo compiler probably guarantees this, but I'd add an assert just in case.